### PR TITLE
Support Qt6

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

**Comment by vkrause** [comment]
This seems to contain a few commits from me that were never intended for integration, but as temporary placeholders, ie. changes that should be replaced by proper solutions, not built on top of. Is this something we can still clean up a bit?

**Comment by Waqar144** [comment]
> This seems to contain a few commits from me that were never intended for integration, but as temporary placeholders, ie. changes that should be replaced by proper solutions, not built on top of. Is this something we can still clean up a bit?

Yes, i have created the PR to start preparing it for master.

File: CMakeLists.txt [hunk 1]
```diff
@@ -397,6 +406,17 @@ if(GAMMARAY_BUILD_DOCS)
   find_package(Qt${Qt_VERSION_MAJOR} NO_MODULE QUIET OPTIONAL_COMPONENTS Help LinguistTools)
 endif()
 
+if(${Qt_VERSION_MAJOR} GREATER_EQUAL 6)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  add_definitions(-DGAMMARAY_QT6_TODO)
```
Comment (vkrause) [inline CMakeLists.txt:412]: Do we still need this? That was supposed to be just for temporary scaffolding during porting.

File: cmake/ECMQueryQmake.cmake [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#
```
Comment (vkrause) [inline cmake/ECMQueryQmake.cmake:1]: This file seems unused? There's also one with the same name in cmake/ECM/modules (with a removed copyright header and using a non-existent include?). Maybe some kind of merge conflict?

File: core/tools/messagehandler/messagehandler.cpp [hunk 3]
```diff
@@ -149,10 +149,10 @@ MessageHandler::MessageHandler(Probe *probe, QObject *parent)
     // install handler directly, catches most cases,
     // i.e. user has no special handler or the handler
     // is created before the QApplication
-    ensureHandlerInstalled();
+//     ensureHandlerInstalled();
     // recheck when eventloop is entered, if the user
     // installs a handler after QApp but before .exec()
-    QMetaObject::invokeMethod(this, "ensureHandlerInstalled", Qt::QueuedConnection);
+//     QMetaObject::invokeMethod(this, "ensureHandlerInstalled", Qt::QueuedConnection);
```
Comment (vkrause) [inline core/tools/messagehandler/messagehandler.cpp:155]: Are these changes intentional or should this be conditional on the Qt version?

File: examples/quick-batching/CMakeLists.txt [hunk 4]
```diff
@@ -1 +1 @@
-qml_lint(quick-batching.qml Slider.qml)
+#qml_lint(quick-batching.qml Slider.qml)
```
Comment (vkrause) [inline examples/quick-batching/CMakeLists.txt:1]: Temporary porting leftover? If that doesn't work with 6 we probably should at least make it conditional on the Qt version. Repeats below.

File: plugins/quickinspector/CMakeLists.txt [hunk 5]
```diff
@@ -40,7 +40,7 @@ if(TARGET Qt::Quick)
       SOURCES ${gammaray_quickinspector_srcs}
     )
 
-    if(NOT Qt5Quick_VERSION VERSION_LESS 5.7)
+    if(NOT Qt5Quick_VERSION VERSION_LESS 5.7 OR TARGET Qt6::Quick)
       target_sources(gammaray_quickinspector PUBLIC ${CMAKE_CURRENT_LIST_DIR}/quickimplicitbindingdependencyprovider.cpp)
```
Comment (vkrause) [inline plugins/quickinspector/CMakeLists.txt:44]: There's build fixes for this file below, but it's taken out for Qt6 here it seems?

File: examples/quick-batching/CMakeLists.txt (hunk 4 above)
Comment (Waqar144) [reply examples/quick-batching/CMakeLists.txt]: done. examples seem to work, so renabled.

File: cmake/ECMQueryQmake.cmake (hunk 2 above)
Comment (Waqar144) [reply cmake/ECMQueryQmake.cmake:1]: yes, removed this one as it was recently moved into `cmake/ECM` directory.

File: plugins/quickinspector/CMakeLists.txt (hunk 5 above)
Comment (Waqar144) [reply plugins/quickinspector/CMakeLists.txt:44]: rather its being added explicitly for Qt6

File: core/tools/messagehandler/messagehandler.cpp (hunk 3 above)
Comment (Waqar144) [reply core/tools/messagehandler/messagehandler.cpp]: done.

**Comment by vkrause** [comment]
Let's get this in then.
